pkg/machine/wsl/machine.go

```diff
@@ -0,0 +1,1093 @@
+       /home/user/.config/systemd/user/default.target.wants/linger-example.service
+`
+
+const wslInstallError = `Could not %s. See previous output for any potential failure details.
+If you can not resolve the issue, and rerunning fails, try the "wsl --install" process
+outlined in the following article:
+
+http://docs.microsoft.com/en-us/windows/wsl/install
+
+`
+
+const wslInstallKernel = "install the WSL Kernel"
+
+const wslOldVersion = `Automatic installation of WSL can not be performed on this version of Windows
+Either update to Build 19041 (or later), or perform the manual installation steps
+outlined in the following article:
+
+http://docs.microsoft.com/en-us/windows/wsl/install\
+
+`
+
+type MachineVM struct {
+	// IdentityPath is the fq path to the ssh priv key
+	IdentityPath string
+	// IgnitionFilePath is the fq path to the .ign file
+	ImageStream string
+	// ImagePath is the fq path to
+	ImagePath string
+	// Name of the vm
+	Name string
+	// SSH port for user networking
+	Port int
+	// RemoteUsername of the vm user
+	RemoteUsername string
+}
+
+type ExitCodeError struct {
+	code uint
+}
+
+func (e *ExitCodeError) Error() string {
+	return fmt.Sprintf("Process failed with exit code: %d", e.code)
+}
+
+// NewMachine initializes an instance of a virtual machine based on the qemu
+// virtualization.
+func NewMachine(opts machine.InitOptions) (machine.VM, error) {
+	vm := new(MachineVM)
+	if len(opts.Name) > 0 {
+		vm.Name = opts.Name
+	}
+
+	// An image was specified
+	if len(opts.ImagePath) > 0 {
+		vm.ImagePath = opts.ImagePath
+	}
+
+	// Assign remote user name. if not provided, use default
+	vm.RemoteUsername = opts.Username
+	if len(vm.RemoteUsername) < 1 {
+		vm.RemoteUsername = defaultRemoteUser
+	}
+
+	// Add a random port for ssh
+	port, err := utils.GetRandomPort()
+	if err != nil {
+		return nil, err
+	}
+	vm.Port = port
+
+	return vm, nil
+}
+
+// LoadByName reads a json file that describes a known qemu vm
+// and returns a vm instance
+func LoadVMByName(name string) (machine.VM, error) {
+	vm := new(MachineVM)
+	vmConfigDir, err := machine.GetConfDir(vmtype)
+	if err != nil {
+		return nil, err
+	}
+	b, err := ioutil.ReadFile(filepath.Join(vmConfigDir, name+".json"))
+	if os.IsNotExist(err) {
+		return nil, errors.Wrap(machine.ErrNoSuchVM, name)
+	}
+	if err != nil {
+		return nil, err
+	}
+	err = json.Unmarshal(b, vm)
+	return vm, err
+}
+
+// Init writes the json configuration file to the filesystem for
+// other verbs (start, stop)
+func (v *MachineVM) Init(opts machine.InitOptions) (bool, error) {
+	cont, err := checkAndInstallWSL(opts)
+	if !cont {
+		appendOutputIfError(opts.ReExec, err)
+		return cont, err
+	}
+
+	homeDir, err := machine.GetUserHome()
+	if err != nil {
+		return false, err
+	}
+	sshDir := filepath.Join(homeDir, ".ssh")
+	v.IdentityPath = filepath.Join(sshDir, v.Name)
+
+	err = downloadDistro(v, opts)
+	if err != nil {
+		return false, err
+	}
+
+	err = writeJSON(v)
```

Comment:
Could you write this as  
```
if err:=writeJSON(v); err != nil {
      return false, err
}
```
Ditto all of the err = lines the same way.

Reply:
will do